pandas/tests/strings/test_split_partition.py

```diff
@@ -35,6 +35,30 @@ def test_split(any_string_dtype):
     tm.assert_series_equal(result, exp)
 
 
+def test_split_regex(any_string_dtype):
+    # GH 43563
+    # explicit regex = True split
+    values = Series("xxxjpgzzz.jpg", dtype=any_string_dtype)
+    result = values.str.split(r"\.jpg", regex=True)
+    exp = Series([["xxxjpgzzz", ""]])
+    tm.assert_series_equal(result, exp)
+
+    # explicit regex = False split
+    result = values.str.split(r"\.jpg", regex=False)
+    exp = Series([["xxxjpgzzz.jpg"]])
+    tm.assert_series_equal(result, exp)
+
+    # non explicit regex split, pattern length == 1
+    result = values.str.split(r".")
+    exp = Series([["xxxjpgzzz", "jpg"]])
+    tm.assert_series_equal(result, exp)
+
+    # non explicit regex split, pattern length != 1
+    result = values.str.split(r".jpg")
+    exp = Series([["xx", "zzz", ""]])
+    tm.assert_series_equal(result, exp)
+
```

Comment:
Can you test that a compiled regex with regex=False raises?

Reply:
sure! done :)

Reply:
I think you have to add a test where a compiled regex is passed with `regex=True`

Reply:
@Dr-Irv thanks for your feedback, I've added one!